component/file-storage/src/main/java/org/exoplatform/commons/file/services/impl/FileServiceImpl.java

```diff
@@ -120,7 +144,11 @@ public FileInfo deleteFile(long id) {
     FileInfo fileInfo = dataStorage.getFileInfo(id);
     if (fileInfo != null) {
       fileInfo.setDeleted(true);
-    }
+      try {
+        listenerService.broadcast(FILE_DELETED_EVENT, fileInfo, null);
+      } catch (Exception e) {
+        LOG.error("Error while broadcasting event: {}", FILE_DELETED_EVENT, e);
+      }    }
```

Comment:
Should this part be after the real delete (dataStorage.updateFileInfo) ?

Reply:
done